lib/looker.ts

```diff
@@ -150,3 +164,92 @@ export async function getCTRPercentData(
     };
   }
 }
+
+/**
+ * Removes any messages inside `data` for ids specified in the removeMessages
+ * array and for ids with substring "test".
+ */
+export function cleanLookerData(data: any) {
+  let cleanData = JSON.parse(JSON.stringify(data)).filter((messageDef: any) => {
+    const removeMessages = ["undefined", "", "n/a", null, "DEFAULT_ID"];
+    return (
+      !removeMessages.includes(
+        messageDef[
+          "messaging_system.metrics__text2__messaging_system_message_id"
+        ],
+      ) &&
+      !messageDef[
+        "messaging_system.metrics__text2__messaging_system_message_id"
+      ]
+        .toLowerCase()
+        .includes("test") &&
+      messageDef["messaging_system.metrics__text2__messaging_system_message_id"]
+        .length > 1
+    );
+  });
+  return cleanData;
+}
+
+/**
+ * Appends any messages from `newLookerData` into `originalData` with an id
+ * that does not already exist in `originalData`. Updates the templates for
+ * any message that need some clean up.
+ *
+ * The message data in `newLookerData` has properties
+ * "messaging_system.metrics__text2__messaging_system_message_id" and
+ * "messaging_system.metrics__string__messaging_system_ping_type" to represent
+ * the message id and template. Before appending these messages into
+ * `originalData`, we must clean up the objects to have properties "id" and
+ * "template" instead, and exclude any other properties that do not currently
+ * provide any value.
+ */
+export function mergeLookerData(originalData: any, newLookerData: any) {
+  for (let i = 0; i < newLookerData.length; i++) {
+    if (
+      !originalData.find(
+        (x: any) =>
```

Comment:
Instead of hoisting it, I defined some variables and left a comment there to improve the readability. I hope it's clean enough now but let me know if you think it'd still benefit from being inside a separate function